solver:
  horizon_steps: 20
  dt: 0.07
  position_weight: [10.0, 10.0, 5.0]
  velocity_weight: [1.0, 1.0, 1.0]
  quaternion_weight: [1.8, 1.8, 1.8, 1.8]
  rate_weight: [2.0, 2.0, 0.22]
  control_weight: [0.5, 0.5, 0.5, 0.5]
  terminal_weight: [5.0, 5.0, 5.0,
                    2.0, 2.0, 2.0,
                    7.8, 7.8, 7.8, 7.8,
                    2.0, 2.0, 2.0]
  regularization: 7.5e-04
  iter_max: 20
  codegen_directory: ~/.cache/rotors_mpc_controller/acados

vehicle:
  mass: 0.68
  inertia: [0.0070, 0.0, 0.0,
            0.0, 0.0070, 0.0,
            0.0, 0.0, 0.0120]
  arm_length: 0.17
  rotor_force_constant: 8.54858e-06
  rotor_moment_constant: 0.016
  motor_min_speed: 50.0
  motor_max_speed: 838.0
  rotor_configuration: '+'

controller:
  thrust_limits: [4.0, 20.0]

world:
  gravity: 9.81

reference:
  frame: world
  default_position: [1.0, 1.0, 1.0]
  default_velocity: [0.0, 0.0, 0.0]
  default_acceleration: [0.0, 0.0, 0.0]
  default_yaw: 0.0

topics:
  state: /hummingbird/ground_truth/odometry
  motor: /hummingbird/command/motor_speed
  reference: /mpc_controller/setpoint

node:
  rate: 60.0
  log_interval: 3.0
